src/components/fields/money-field/money-field.js

```diff
@@ -111,7 +122,20 @@ class MoneyField extends React.Component {
             description={this.props.description}
             onInfoButtonClick={this.props.onInfoButtonClick}
             hintIcon={this.props.hintIcon}
-            badge={this.props.badge}
+            badge={
+              this.props.showHighPrecisionBadge &&
```

Comment:
`shouldShowHighPrecisionBadge`. I know it gets long. Maybe these prefixes for React props are something we should really drop.

Reply:
Should I do it or not? I thought the `show`  prefix is as a good enough indicator. But it does sound like a function now that you mention it.

Reply:
Maybe just `hasHighPrecisionBadge`. The potentially confusing part is that the badge will only be shown when the value holds a high precision price. But the `show` prefix has the same problem.